### ISSUE TITLE
Cannot get bibtexparser .loads to work (__call__ missing argument at bibtexexpression.py, line 180)

### ISSUE THREAD

**Issue by sdaau** [sciunto-org/python-bibtexparser#215]
I think I just installed bibtexparser today, on a Ubuntu 14.04.5 LTS system, with:

```
sudo -H pip2 install bibtexparser
sudo -H pip3 install bibtexparser
```
Then, I try the exact example from https://bibtexparser.readthedocs.io/en/master/logging.html - here as `test4.py`:

```
#!/usr/bin/python
# -*- coding: utf-8 -*-

import logging
import logging.config

logger = logging.getLogger(__name__)

logging.config.dictConfig({
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'standard': {
            'format': '%(asctime)s [%(levelname)s] %(name)s %(funcName)s:%(lineno)d: %(message)s'
        },
    },
    'handlers': {
        'default': {
            'level':'DEBUG',
            'formatter': 'standard',
            'class':'logging.StreamHandler',
        },
    },
    'loggers': {
        '': {
            'handlers': ['default'],
            'level': 'DEBUG',
            'formatter': 'standard',
            'propagate': True
        }
    }
})


if __name__ == '__main__':
    bibtex = """@ARTICLE{Cesar2013,
      author = {Jean César},
      title = {An amazing title},
      year = {2013},
      month = jan,
      volume = {12},
      pages = {12--23},
      journal = {Nice Journal},
      abstract = {This is an abstract. This line should be long enough to test
             multilines...},
      comments = {A comment},
      keywords = {keyword1, keyword2},
    }
    """

    with open('/tmp/bibtex.bib', 'w') as bibfile:
        bibfile.write(bibtex)

    from bibtexparser.bparser import BibTexParser

    with open('/tmp/bibtex.bib', 'r') as bibfile:
        bp = BibTexParser(bibfile.read())
        print(bp.get_entry_list())
```

Both python 2 and 3 fail with:

```
$ python2 test4.py
Traceback (most recent call last):
  File "test4.py", line 57, in <module>
    bp = BibTexParser(bibfile.read())
  File "/usr/local/lib/python2.7/dist-packages/bibtexparser/bparser.py", line 75, in __new__
    return parse(data, **args)
  File "/usr/local/lib/python2.7/dist-packages/bibtexparser/bparser.py", line 30, in parse
    parser = BibTexParser(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/bibtexparser/bparser.py", line 132, in __init__
    self._init_expressions()
  File "/usr/local/lib/python2.7/dist-packages/bibtexparser/bparser.py", line 171, in _init_expressions
    self._expr = BibtexExpression()
  File "/usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py", line 180, in __init__
    ) | pp.stringEnd()
TypeError: __call__() takes exactly 2 arguments (1 given)

$ python3 test4.py
Traceback (most recent call last):
  File "test4.py", line 57, in <module>
    bp = BibTexParser(bibfile.read())
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bparser.py", line 75, in __new__
    return parse(data, **args)
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bparser.py", line 30, in parse
    parser = BibTexParser(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bparser.py", line 132, in __init__
    self._init_expressions()
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bparser.py", line 171, in _init_expressions
    self._expr = BibtexExpression()
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bibtexexpression.py", line 180, in __init__
    ) | pp.stringEnd()
TypeError: __call__() missing 1 required positional argument: 'name'
```

In fact, I don't even need all that much code, tried this as `test3.py`:

```
#!/usr/bin/python
# -*- coding: utf-8 -*-

import os, sys
import re
import bibtexparser

bib_database = bibtexparser.loads("")
```

... and I get the exact same error, at `bibtexparser/bibtexexpression.py`, line 180 ?!

Not sure if this is a packaging bug, or I installed wrongly somehow...

**Comment by sdaau** [comment]
Ok, I think I found the problem - it seems it is a typo in the code of `bibtexexpression.py`; here is the section around line 180 that I had originally (it's the same file for both python 2 and 3):

```
$ less -N /usr/local/lib/python3.4/dist-packages/bibtexparser/bibdatabase.py
...
    177         # Explicit comments: @comment + everything up to next valid declaration
    178         # starting on new line.
    179         not_an_implicit_comment = (pp.LineStart() + pp.Literal('@')
    180                                    ) | pp.stringEnd()
    181         self.explicit_comment = (

```

Here's the bit of debugging I did:

```
$ pdb test3.py
> /tmp/test3.py(4)<module>()
-> import os, sys
(Pdb) b /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py:172
Breakpoint 1 at /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py:172
(Pdb) r
> /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py(172)__init__()
-> in_braces_or_pars(key + pp.Suppress(',') + field_list)
(Pdb) n
> /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py(173)__init__()
-> )('Entry')
(Pdb) 
> /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py(179)__init__()
-> not_an_implicit_comment = (pp.LineStart() + pp.Literal('@')
(Pdb) p pp
<module 'pyparsing' from '/usr/lib/python2.7/dist-packages/pyparsing.pyc'>
(Pdb) p pp.LineStart()
LineStart
(Pdb) p pp.Literal('@')
"@"
(Pdb) n
> /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py(180)__init__()
-> ) | pp.stringEnd()
(Pdb) 
TypeError: '__call__() takes exactly 2 arguments (1 given)'
> /usr/local/lib/python2.7/dist-packages/bibtexparser/bibtexexpression.py(180)__init__()
-> ) | pp.stringEnd()
(Pdb) p pp.stringEnd()
*** TypeError: TypeError('__call__() takes exactly 2 arguments (1 given)',)
(Pdb) p pp.StringEnd()
StringEnd
(Pdb) 
```

So it turns out it is a typo - in line 180, instead of `pp.stringEnd()`, one should have `pp.StringEnd()` (I made the same change in both the python 2 and 3 library versions of the `bibtexexpression.py` file).

With this, my "empty string" `test3.py` example above passes with python 2, but python 3 seems to not allow it anymore:

```
$ python3 test3.py 
Traceback (most recent call last):
  File "test3.py", line 8, in <module>
    bib_database = bibtexparser.loads("")
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/__init__.py", line 48, in loads
    return parser.parse(bibtex_str)
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bparser.py", line 145, in parse
    bibtex_file_obj = self._bibtex_file_obj(bibtex_str)
  File "/usr/local/lib/python3.4/dist-packages/bibtexparser/bparser.py", line 213, in _bibtex_file_obj
    if bibtex_str[0] == byte:
IndexError: string index out of range
```

.... while, the `test4.py` example is made to generate an error, but that can be fixed with adding `common_strings=True` here:

```
        # bp = BibTexParser(bibfile.read()) # deliberately causes error;
        bp = BibTexParser(bibfile.read(), common_strings=True)
```

... and in that case, both python 2 and 3 work fine:

```
$ python2 test4.py 
2018-08-22 01:16:43,428 [DEBUG] bibtexparser.bparser action:31: Found Entry: [u'ARTICLE', u'Cesar2013', {u'title': u'An amazing title', u'abstract': u'This is an abstract. This line should be long enough to test\nmultilines...', u'author': u'Jean C\xe9sar', u'comments': u'A comment', u'pages': u'12--23', u'volume': u'12', u'year': u'2013', u'keywords': u'keyword1, keyword2', u'month': 'January', u'journal': u'Nice Journal'}]
[{'ENTRYTYPE': u'article', u'author': u'Jean C\xe9sar', u'journal': u'Nice Journal', u'abstract': u'This is an abstract. This line should be long enough to test\nmultilines...', u'title': u'An amazing title', u'comments': u'A comment', u'month': 'January', u'volume': u'12', u'year': u'2013', u'keywords': u'keyword1, keyword2', 'ID': u'Cesar2013', u'pages': u'12--23'}]

$ python3 test4.py 
2018-08-22 01:16:44,880 [DEBUG] bibtexparser.bparser action:31: Found Entry: ['ARTICLE', 'Cesar2013', {'author': 'Jean César', 'pages': '12--23', 'year': '2013', 'volume': '12', 'journal': 'Nice Journal', 'keywords': 'keyword1, keyword2', 'title': 'An amazing title', 'abstract': 'This is an abstract. This line should be long enough to test\nmultilines...', 'comments': 'A comment', 'month': 'January'}]
[{'pages': '12--23', 'year': '2013', 'author': 'Jean César', 'volume': '12', 'month': 'January', 'journal': 'Nice Journal', 'abstract': 'This is an abstract. This line should be long enough to test\nmultilines...', 'ENTRYTYPE': 'article', 'ID': 'Cesar2013', 'keywords': 'keyword1, keyword2', 'comments': 'A comment', 'title': 'An amazing title'}]
```

**Comment by omangin** [comment]
@sdaau, thanks for reporting this. I still have to look at it more closely, though.

**Comment by nvmnghia** [comment]
I had the same bug, using Python 2 venv!
When I ran the code in terminal, it executed without error. When I use IntelliJ with python plugin and venv, this bug show up and I edited just like @sdaau . There's no stringEnd() function anyway.

**Comment by omangin** [comment]
It seems that this was fixed by #209. Thanks for reporting anyway.

Please reopen if the issue is still there.
